File: server/python/calculadora.py
def soma(a, b):
    return a + b


def subtracao(a, b):
    return a - b


def multiplicacao(a, b):
    return a * b


def divisao(a, b):
    if b == 0:
        return "Erro: divisão por zero"
    return a / b


def main():
    operacao = input("Escolha a operação (+, -, *, /): ")

    a = float(input("Digite o valor 1: "))
    b = float(input("Digite o valor 2: "))

    if operacao == "+":
        print(f"Resultado: {soma(a,b)}")
    elif operacao == "-":
        print(f"Resultado: {soma(a,b)}")
    elif operacao == "*":
        print(f"Resultado: {soma(a,b)}")
    elif operacao == "/":
        resultado = divisao(a, b)
        if isinstance(resultado, str):
            print(resultado)
        else:
            print(f"Resultado: {resultado:.2f}")
    else:
        print("Operação inválida!")


if __name__ == "__main__":
    main()

File: server/c/calculadora.c
#include <stdio.h>

double soma(double a, double b)
{
    return a + b;
}

double subtracao(double a, double b)
{
    return a - b;
}

double multiplicacao(double a, double b)
{
    return a * b;
}

double divisao(double a, double b)
{
    if (b == 0)
    {
        printf("Erro: divisão por zero");
    }
    return a / b;
}

int main()
{
    double a, b;
    char operacao;

    printf("Escolha a operação (+, -, *, /): ");
    scanf("%c", &operacao);

    printf("Digite o valor 1: ");
    scanf("%lf", &a);

    printf("Digite o valor 2: ");
    scanf("%lf", &b);

    if (operacao == '+')
    {
        printf("Resultado: %.2lf\n", soma(a, b));
    }
    else if (operacao == '-')
    {
        printf("Resultado: %.2lf\n", subtracao(a, b));
    }
    else if (operacao == '*')
    {
        printf("Resultado: %.2lf\n", multiplicacao(a, b));
    }
    else if (operacao == '/')
    {
        printf("Resultado: %.2lf\n", divisao(a, b));
    }
    else
    {
        printf("Digite um numero correto para a operacao!");
    }

    return 0;
}